autotest/t076_modelgrid_thick.py

```diff
@@ -0,0 +1,196 @@
+"""
+Model grid thick method tests
+
+"""
+
+import numpy as np
+from flopy.discretization import StructuredGrid, VertexGrid, UnstructuredGrid
+
+def test_structured_thick():
+    nlay, nrow, ncol = 3, 2, 3
+    delc = 1. * np.ones(nrow, dtype=float)
+    delr = 1. * np.ones(ncol, dtype=float)
+    top = 10. * np.ones((nrow, ncol), dtype=float)
+    botm = np.zeros((nlay, nrow, ncol), dtype=float)
+    botm[0, :, :] = 5.
+    botm[1, :, :] = 0.
+    botm[2, :, :] = -5.
+    grid = StructuredGrid(
+        nlay=nlay,
+        nrow=nrow,
+        ncol=ncol,
+        delc=delc,
+        delr=delr,
+        top=top,
+        botm=botm,
+    )
+    thick = grid.thick()
+    assert np.allclose(thick, 5.), "thicknesses != 5."
+
+    sat_thick = grid.thick(array=grid.botm + 10.)
+    assert np.allclose(sat_thick, thick), "saturated thicknesses != 5."
+
+    sat_thick = grid.thick(array=grid.botm + 5.)
+    assert np.allclose(sat_thick, thick), "saturated thicknesses != 5."
+
+    sat_thick = grid.thick(array=grid.botm + 2.5)
+    assert np.allclose(sat_thick, 2.5), "saturated thicknesses != 2.5"
+
+    sat_thick = grid.thick(array=grid.botm)
+    assert np.allclose(sat_thick, 0.), "saturated thicknesses != 0."
+
+    sat_thick = grid.thick(array=grid.botm-100.)
+    assert np.allclose(sat_thick, 0.), "saturated thicknesses != 0."
+
+    return
+
+def test_vertices_thick():
+    nlay, ncpl = 3, 5
+    vertices = [
+        [0, 0.0, 3.0],
+        [1, 1.0, 3.0],
+        [2, 2.0, 3.0],
+        [3, 0.0, 2.0],
+        [4, 1.0, 2.0],
+        [5, 2.0, 2.0],
+        [6, 0.0, 1.0],
+        [7, 1.0, 1.0],
+        [8, 2.0, 1.0],
+        [9, 0.0, 0.0],
+        [10, 1.0, 0.0],
+    ]
+    iverts = [
+        [0, 0, 1, 4, 3],
+        [1, 1, 2, 5, 4],
+        [2, 3, 4, 7, 6],
+        [3, 4, 5, 8, 7],
+        [4, 6, 7, 10, 9],
+        [5, 0, 1, 4, 3],
+        [6, 1, 2, 5, 4],
+        [7, 3, 4, 7, 6],
+        [8, 4, 5, 8, 7],
+        [9, 6, 7, 10, 9],
+        [10, 0, 1, 4, 3],
+        [11, 1, 2, 5, 4],
+        [12, 3, 4, 7, 6],
+        [13, 4, 5, 8, 7],
+        [14, 6, 7, 10, 9],
+    ]
+    top = np.ones((nlay, ncpl), dtype=float)
+    top[0, :] = 10.
+    top[1, :] = 5.
+    top[2, :] = 0.
```

Comment:
```suggestion
    top = np.ones((ncpl, ), dtype=float) * 10
```
Vertex grid follows the Modflow 6 DISV input style of only providing top for the Model top